ui/src/app/store/machine/slice.ts

```diff
@@ -28,6 +29,24 @@ export type ScriptInput = {
   [x: string]: { url: string };
 };
 
+const generateParams = <P extends { [x: string]: unknown }>(
```

Comment:
Nice. Is the plan to have a type for each action's params? Might be time to break up the types file :grimacing:

Reply:
That could be helpful, but this at least should remove the need for all of the optional param spreads e.g.
```typescript
...("partitionIds" in params && {
  add_partitions: params.partitionIds,
}),
```

Reply:
Yep that was getting a bit unwieldy.